sled-agent/src/common/instance.rs

```diff
@@ -115,15 +115,40 @@ impl ObservedPropolisState {
     /// Propolis.
     pub fn new(
         instance_runtime: &InstanceRuntimeState,
+        propolis_id: PropolisUuid,
         propolis_state: &InstanceStateMonitorResponse,
     ) -> Self {
+        // Decide which of Propolis's reported migrations to pay attention to.
+        // If this Propolis is currently a migration target (evidenced by its ID
+        // appearing in the `dst_propolis_id` field in the instance record),
+        // look at the current migration in. Otherwise, look at the reported
+        // migration out.
+        let role = if instance_runtime
+            .dst_propolis_id
+            .is_some_and(|id| id == propolis_id)
+        {
+            MigrationRole::Target
+        } else {
+            MigrationRole::Source
+        };
```

Comment:
This logic is going to have to change substantially once sled-agent no longer has an `InstanceRuntimeState` to make this determination. I think we ought to be able to do this instead using the `migration_state` in `InstanceStates`, which should also have the ID of the currently active migration (which would let us check whether the migration in or migration out refers to the current migration).

Reply:
Good suggestion, thanks! d9b4ae8 should clean this up a bit.